docs/source/contributor-guide/communication.md

```diff
@@ -68,6 +68,11 @@ Unfortunately, due to spammers, the ASF Slack workspace requires an invitation
 to join. We are happy to invite you -- please ask for an invitation in the
 Discord server.
 
+### Job Board
+
+There are plenty of opportunities to work on DataFusion advertised on the
+#hiring channel on the [Arrow Rust Discord Server][discord-link].
```

Comment:
I think it would be nice to also make it clear prospective employers should feel free to post there as well. Perhaps someting like:

```suggestion
#hiring channel on the [Arrow Rust Discord Server][discord-link].
Please feel free to post links to DataFusion related jobs there. 
```